.github/workflows/build_steps.yml

```diff
@@ -119,17 +119,45 @@ jobs:
           configurePresetAdditionalArgs: "['-DVCPKG_INSTALL_OPTIONS=--clean-after-build']"
           buildPreset: ${{env.ARCTIC_CMAKE_PRESET}}
 
+      - name: Select Python (Linux)
+        if: inputs.job_type == 'leader-compile' && matrix.os == 'linux'
+        run: echo /opt/python/cp36-cp36m/bin >> $GITHUB_PATH
+          
+      - name: Test with pytest
+        if: inputs.job_type == 'leader-compile' && matrix.os == 'linux'
+        run: |
+          yum install nodejs npm -y
+          npm config set strict-ssl false
+          npm install -g azurite
+          python -m pip install --upgrade pip
+
+          python >build_tooling/requirements-analysis-workflow.txt <<END
```

Comment:
This will be the third copy of this code :(

Reply:
yeah if we agree to do it this way, I will refactor it to a local action, similar to the one for the persistent storage vars